lib/uniris/mining/proof_of_existence.ex

```diff
@@ -0,0 +1,35 @@
+defmodule Uniris.Mining.ProofOfExistence do
+  @moduledoc false
+
+  alias Uniris.Oracles.TransactionContent
+
+  alias Uniris.TransactionChain.{
+    Transaction,
+    TransactionData
+  }
+
+  # Public
+
+  @spec do_proof_of_existence(Transaction.t()) :: boolean()
+  def do_proof_of_existence(tx) do
+    %TransactionData{content: content} = tx.data
+    %TransactionContent{mfa: {m, f, a}, payload: payload} = :erlang.binary_to_term(content)
```

Comment:
I think we should avoid `binary_to_term` as it's not safe and some malicious code can be injected. We can easily provide a serialization mechanism using binaries and bitstrings adapted for the given oracle.